pkg/fake/types.go

```diff
@@ -51,6 +60,23 @@ func (m *MockedFunction[I, O]) Invoke(input *I, defaultTransformer func(*I) (*O,
 		m.successfulCalls.Add(1)
 		return m.Output.Clone(), nil
 	}
+	// This output pages multi-threaded handling isn't perfect
+	// It will fail if pages are asynchronously requested from the same NextToken
```

Comment:
Just sanity checking, but this should never happen in our functional tests right? This is just a callout for the future?

Reply:
Right